pkg/api/validation/validation.go

```diff
@@ -2754,10 +2759,22 @@ func ValidateResourceRequirements(requirements *api.ResourceRequirements, fldPat
 		if api.IsStandardResourceName(string(resourceName)) {
 			allErrs = append(allErrs, validateBasicResource(quantity, fldPath.Key(string(resourceName)))...)
 		}
+		if resourceName == api.ResourceMemory && !IsResourceRequirementsMemoryValid(quantity) {
+			allErrs = append(allErrs, field.Invalid(fldPath, quantity.String(), "memory request must not be configured in millibytes"))
+		}
 	}
 	return allErrs
 }
 
+// validateResourceRequirementsMemory ensures that memory is not configured in milli value
+func IsResourceRequirementsMemoryValid(quantity resource.Quantity) bool {
+	memStr := quantity.String()
+	if memStr[len(memStr)-1] == 'm' {
```

Comment:
This is not the right way to test it. You need to check that rounding to whole units doesn't change the value.